apps/prairielearn/src/lib/course-request.ts

```diff
@@ -0,0 +1,107 @@
+import { loadSqlEquiv, queryRow, queryAsync } from '@prairielearn/postgres';
+import { z } from 'zod';
+import { logger } from '@prairielearn/logger';
+import * as Sentry from '@prairielearn/sentry';
+
+import { DateFromISOString, IdSchema } from '../lib/db-types';
+import { createCourseRepoJob } from '../lib/github';
+import { sendCourseRequestMessage } from '../lib/opsbot';
+
+const sql = loadSqlEquiv(__filename);
+
+const JobsRowSchema = z
+  .object({
+    authn_user_id: IdSchema.nullable(),
+    authn_user_name: z.string().nullable(),
+    finish_date: DateFromISOString.nullable(),
+    id: IdSchema,
+    number: z.number(),
+    start_date: DateFromISOString,
+    status: z.string(),
+  })
+  .optional();
```

Comment:
Why is this marked optional? Generally speaking, top-level schemas like this should just be `z.object(...)`; if they might be optional in other contexts, you can do something like `jobs: z.array(JobsRowSchema.optional())`. That said, I can't see a case where we'd end up with undefined entries in the `jobs` array.

Reply:
That makes sense. Originally, I was thinking we would need `.optional()` in the case of an empty array but that is not the case. I have removed it here and will only add it inline to where it is being used in the future.